pandas/_libs/window.pyx

```diff
@@ -1349,8 +1349,6 @@ cdef _roll_min_max_variable(ndarray[numeric] values,
 
 
 cdef _roll_min_max_fixed(ndarray[numeric] values,
-                         ndarray[int64_t] starti,
```

Comment:
why exactly would this cause a memory leak? if it did then we would be leaking in many parts of pandas as this idiom is used a lot.

Reply:
The key here is that those args aren't used at all in the function.  If the args are used, they are garbage collected properly.

I have reproduced this outside of pandas, and it happens with the latest numpy, so it appears to be a numpy issue.  If you want I can share my files that I used for reproducing it, or I can open a bug report with numpy as well.

Reply:
ok thanks for the explanation

i would open an issue with numpy as well if u can make a minimal repro